go-controller/pkg/ovn/pods.go

```diff
@@ -381,5 +386,46 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 	// observe the pod creation latency metric.
 	recordPodCreated(pod)
 
+	if err = oc.addLogicalSwitchPodCount(logicalSwitchInfo, logicalSwitch); err != nil {
+		return fmt.Errorf("failed to update pod count for the logical switch %s for pod %s - %v",
+			logicalSwitch, pod.Name, err)
+	}
+
+	return nil
+}
+
+// addLogicalSwitchPodCount updates the per node metric for ip utilization.
+// It also increments the podCount cache for the switch given by nodeName.
+func (oc *Controller) addLogicalSwitchPodCount(lsInfo *logicalSwitchInfo, nodeName string) error {
+	if lsInfo == nil {
+		return fmt.Errorf("Invalid logical switch info while adding pod count for node %s", nodeName)
+	}
+	if nodeName == "" {
+		return fmt.Errorf("Invalid node name provided while incrementing pod count")
+	}
+	lsInfo.podCount = lsInfo.podCount + 1
```

Comment:
This function is called exactly from one place and from where it is called `lsInfo` and `nodename` already have right values. So, you can remove lines 400-405 and this function shouldn't return `error`